File: backend/src/main/java/com/example/HopitalPlanningProject/model/Planning.java
package com.example.HopitalPlanningProject.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Planning {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idPlanning; // Primary Key

    private byte nbSemaineMax;
    private byte nbRoulementMax;
    private int nbJoursMaxTravailConsecutifs;
    private int nbJoursMinTravailConsecutifs;
    private int frequenceWeekend;
}

File: frontend/src/models/Planning.js
// models/Planning.js
class Planning {
    constructor(idPlanning, nbSemaineMax, nbRoulementMax, nbJoursMaxTravailConsecutifs, nbJoursMinTravailConsecutifs, frequenceWeekend) {
        this.idPlanning = idPlanning;
        this.nbSemaineMax = nbSemaineMax;
        this.nbRoulementMax = nbRoulementMax;
        this.nbJoursMaxTravailConsecutifs = nbJoursMaxTravailConsecutifs;
        this.nbJoursMinTravailConsecutifs = nbJoursMinTravailConsecutifs;
        this.frequenceWeekend = frequenceWeekend;
    }
}

export default Planning;
